### PR TITLE
fix: replace custom regex password validation with Django's built-in validators

### LINKED ISSUE #3385: Bug: REGEX_PASSWORD allows weak passwords
###  What happened
I was going through the codebase and noticed that the password validation regex in intel_owl/consts.py is way too permissive. Here's what it currently looks like:
```
# intel_owl/consts.py
REGEX_PASSWORD = r"^[a-zA-Z0-9]{12,}$"
```
This regex only checks two things:

The password is at least 12 characters long
It only contains letters and numbers
That's it. No uppercase requirement, no lowercase requirement, no digit requirement, no special characters. And because of the [a-zA-Z0-9] character class, it actually blocks special characters entirely — so even users who want to use a stronger password with symbols like !, @, or # can't.

and the issue is regex is not just being in the code but it's wired directly into the user registration flow in authentication/serializers.py:
```
# /authentication/serializers.py
# valid password defination
def validate_password(self, password):
    super().validate_password(password)

    if re.match(REGEX_PASSWORD, password):
        return password
    else:
        raise ValidationError("Invalid password")
```
IntelOwl is a threat intelligence platform where analysts store API keys, configure external service integrations, and handle sensitive investigation data. Weak passwords on accounts like these are a real problem.

### Environment
1. OS: Any
2. IntelOwl version: Latest master

### What did you expect to happen
The password validation should actually enforce some complexity — at minimum, requiring a mix of uppercase, lowercase, digits, and ideally at least one special character. A password like aaaaaaaaaaaa should be rejected, not accepted.

Also, the current regex blocks special characters entirely. A user trying to set **pasword@123#** would be told their password is invalid, which is backwards — the stronger password gets rejected while the weaker one sails through.

### How to reproduce your issue
1. Register a new account on IntelOwl.
2. Try setting the password to aaaaaaaaaaaa (12 identical lowercase letters).
3. It passes validation and the account is created successfully.
4. Now create another accound use hard password with special character it says invalid.

### Suggested fix:

Update the regex to require at least one character from each class, and allow (not block) special characters

@mlodic assign this issue i have created a solution on my own too i just want to implement it assign it so i can create a pr.

### PR DESCRIPTION
**Pull request by HarshitPal25**
## Description
This PR replaces the overly permissive `REGEX_PASSWORD` with Django's built-in `AUTH_PASSWORD_VALIDATORS` to enforce robust password policies while allowing secure special characters. This addresses the feedback provided in the previously closed PRs and cleanly resolves the issue.

### Changes Included:
- **Removed `REGEX_PASSWORD`**: Eliminated the restrictive custom regex from `intel_owl/consts.py` that was blocking valid special characters.
- **Implemented Django Validators**: Updated `validate_password_strength` to utilize `django.contrib.auth.password_validation.validate_password`.
- **Configured `MinimumLengthValidator`**: Ensured backward compatibility strictly maintaining the `min_length=12` policy natively via `intel_owl/settings/auth.py`. 
- **Passed User Context for Similarity Validation**: Properly instantiated a temporary `User` object during registration inside `RegistrationSerializer` and updated `ChangePasswordView` so `UserAttributeSimilarityValidator` correctly blocks passwords that are identical or far too similar to the user's username/email.
- **Improved View Error Handling**: Added explicit null checks for `old_password` and `new_password` in `ChangePasswordView` and normalized the error message strings to adhere strictly to the project’s standard JSON error response formatting.
- **Comprehensive Test Suite Updates**: 
  - Passed all existing tests cleanly.
  - Added coverage for missing `old_password`/`new_password` edge cases.
  - Modified tests to accurately reflect that special characters are now allowed and similarity checks/common passwords will rightfully return a `400` validation error response.

Closes #3385

### CODE REVIEW

**Comment by HarshitPal25** [comment]
Hi @mlodic,

I apologize for my previous pushy communication and I understand that I should have waited for assignment before submitting PRs in the past. 

I've successfully rewritten the password validation implementation locally to fully address your feedback and the constraints of `UserAttributeSimilarityValidator` (as well as handling null checks and passing your CI tests on the latest develop branch).

Thank you for your patience and for guiding me through the contributing guidelines.

File: intel_owl/consts.py [hunk 1]
```diff
@@ -1,19 +1,29 @@
 # This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
 # See the file 'LICENSE' for copying permission.
 
-import re
-
+from django.contrib.auth.password_validation import validate_password as django_validate_password
 from django.core.exceptions import ValidationError
```
[the remark this answers is not in the document]
Comment (HarshitPal25) [reply intel_owl/consts.py:6]: @copilot apply changes based on [this feedback](https://github.com/intelowlproject/IntelOwl/pull/3645#discussion_r3057888176)

File: intel_owl/consts.py [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
+    - CommonPasswordValidator
+    - NumericPasswordValidator
+
+    Raises ValidationError if password is invalid.
+    """
+    try:
+        django_validate_password(password, user=user)
+    except ValidationError as e:
+        # Re-raise with combined error messages
+        raise ValidationError("; ".join(e.messages))
```
[the remark this answers is not in the document]
Comment (HarshitPal25) [reply intel_owl/consts.py:26]: @copilot apply changes based on [this feedback](https://github.com/intelowlproject/IntelOwl/pull/3645#discussion_r3057888220)

File: authentication/views.py [hunk 3]
```diff
@@ -184,14 +184,22 @@ def post(request: Request) -> Response:
...
+            return Response({"error": "New password is required"}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Validate new password strength with user context for UserAttributeSimilarityValidator
+        user = request.user
         try:
-            validate_password_strength(new_password)
+            validate_password_strength(new_password, user=user)
         except ValidationError as e:
-            return Response({"error": e.message}, status=status.HTTP_400_BAD_REQUEST)
+            return Response({"error": str(e.message)}, status=status.HTTP_400_BAD_REQUEST)
```
[the remark this answers is not in the document]
Comment (HarshitPal25) [reply authentication/views.py:200]: @copilot apply changes based on [this feedback](https://github.com/intelowlproject/IntelOwl/pull/3645#discussion_r3057888265)

**Comment by HarshitPal25** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/intelowlproject/IntelOwl/pull/3645#pullrequestreview-4082370047)

**Comment by HarshitPal25** [comment]
@mlodic i have make the changes and also read and take the suggested changes by the copilot could you review my pr.

**Comment by mlodic** [comment]
please provide screenshots of how the result of this improvement work
